spec/serializers/site_serializer_spec.rb

```diff
@@ -192,6 +192,27 @@
     end
   end
 
+  describe "#anonymous_sidebar_sections" do
+    fab!(:user) { Fabricate(:user) }
+    fab!(:public_sidebar_section) do
+      Fabricate(:sidebar_section, title: "Public section", user: user, public: true)
```

Comment:
I'm a little confused by the `user` association for a public sidebar section. Since it is a public sidebar section, there should be no need to associate it with a user. Perhaps we need a `created_by_user_id`  column here instead if we really need to know the user that created the public section?

Reply:
Fair, technically, we have StaffAction so don't need information about author for public

Reply:
It would require a bit more of refactoring. SidebarSectionLinks are inheriting user from SidebarSection

https://github.com/discourse/discourse/blob/main/app/models/sidebar_section_link.rb#L15

It can be done - do you think it is worth at this point?

Reply:
> SidebarSectionLinks are inheriting user from SidebarSection

How does an admin create a custom sidebar section? Do they do it through an admin interface of some sort where all admins are able to alter a public custom sidebar section? Currently, we assume that all links in a sidebar section belongs to the same user. This might become harder to untangle in the future.